src/test/java/edu/wpi/first/outlineviewer/controller/ConnectionIndicatorControllerTest.java

```diff
@@ -1,39 +1,95 @@
 package edu.wpi.first.outlineviewer.controller;
 
-import edu.wpi.first.outlineviewer.FxHelper;
+import edu.wpi.first.networktables.NetworkTableInstance;
 import edu.wpi.first.outlineviewer.NetworkTableUtilities;
 import edu.wpi.first.outlineviewer.OutlineViewer;
+import edu.wpi.first.outlineviewer.Preferences;
+import java.util.stream.Stream;
 import javafx.fxml.FXMLLoader;
 import javafx.scene.Scene;
 import javafx.scene.control.Label;
 import javafx.scene.layout.Pane;
 import javafx.stage.Stage;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.testfx.framework.junit5.ApplicationTest;
+import org.testfx.util.WaitForAsyncUtils;
 
 import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 public class ConnectionIndicatorControllerTest extends ApplicationTest {
 
   private ConnectionIndicatorController controller;
+  private NetworkTableInstance mockNetworkTableInstance;
 
   @Override
   public void start(Stage stage) throws Exception {
     NetworkTableUtilities.shutdown();
+
+    mockNetworkTableInstance = mock(NetworkTableInstance.class);
```

Comment:
I mean, sure, personally, I would have created an adapter interface that, in production called directly to the `NetworkTableInstance` but in test I could create my own mock instance of it. But this works too.

If you create the adapter though, you can reuse it in other places in the code.